### PR TITLE
[107] Add ability for users to accept group invitations

### LINKED ISSUE #107: Students can Accept Group Invitations
blocked on #90

### PR DESCRIPTION
**Pull request by esoterik**
Resolves #107

### CODE REVIEW

File: spec/features/groups/accept_invitation_spec.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+require 'rails_helper'
+
+RSpec.feature 'Students Joining Groups' do
+  context 'accepting invitation' do
```
Comment (orenyk) [inline spec/features/groups/accept_invitation_spec.rb]: Why isn't this the top-level description? It seems silly unless you want to group it with sending invitations?

File: spec/policies/group_policy_spec.rb [hunk 2]
```diff
@@ -28,6 +28,22 @@
...
     end
+    permissions :accept_invitation? do
+      context 'invited, not in group' do
+        before do
+          allow(other_group).to receive(:invitations).and_return([user])
+          allow(user).to receive(:group).and_return(nil)
+        end
+        it { is_expected.to permit(user, other_group) }
+      end
+      context 'in draw, in group' do
```
Comment (orenyk) [inline spec/policies/group_policy_spec.rb]: What do you mean by `in draw`?

**Comment by orenyk** [comment]
It does still need to be rebased.

**Comment by esoterik** [comment]
rebased and ready to go

**Pull request merged**